src/peer_manager.rs

```diff
@@ -560,7 +542,17 @@ impl PeerManager {
                 if tunnel {
                     Err(RoutingError::CandidateIsTunnelling)
                 } else {
-                    Ok(Some((candidate.seed.clone(), candidate.target_size)))
+                    let (target_size, seed) = if difficulty == 0 {
+                        (10, vec![5u8; 4])
+                    } else {
+                        let evaluators = cmp::max(self.routing_table.min_section_size(),
+                                                  self.routing_table.our_section().len());
+                        // TODO: the algorithm need to be measured and refactored against the real network
+                        (2 * 1024 * 1024 / evaluators,
+                         rand::thread_rng().gen_iter().take(10).collect())
+                    };
+                    candidate.resource_proof = Some((target_size, difficulty, seed.clone()));
```

Comment:
if section 0, having 20 nodes, being asked to merge with section 1 having 7 nodes and min_section_size is 8.
if the joining node receives get_node_name response from section 0, then starts connecting, but section 0 now merged with section 1, calculation here will use a bigger sized own_section, which results about 33% smaller target_size.
This is just a tiny edge case, and the network shall be ok to stay with it.